app/src/main/java/com/hynekbraun/openmeteoweather/presentation/mainscreen/WeatherViewModel.kt

```diff
@@ -0,0 +1,106 @@
+package com.hynekbraun.openmeteoweather.presentation.mainscreen
+
+import android.location.Location
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hynekbraun.openmeteoweather.domain.CurrentLocationManager
+import com.hynekbraun.openmeteoweather.domain.LocationError
+import com.hynekbraun.openmeteoweather.domain.WeatherFetchError
+import com.hynekbraun.openmeteoweather.domain.WeatherRepository
+import com.hynekbraun.openmeteoweather.domain.mapper.toCurrentData
+import com.hynekbraun.openmeteoweather.domain.mapper.toCurrentHourlyForecastData
+import com.hynekbraun.openmeteoweather.domain.mapper.toDailyForecastData
+import com.hynekbraun.openmeteoweather.domain.util.Resource
+import com.hynekbraun.openmeteoweather.presentation.mainscreen.util.*
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class WeatherViewModel @Inject constructor(
+    private val repository: WeatherRepository,
+    private val currentLocation: CurrentLocationManager
+) : ViewModel() {
+
+    var weatherState by mutableStateOf(WeatherState())
+        private set
+
+    private val eventChannel = Channel<ToastEventHandler>()
+    val eventFlow = eventChannel.receiveAsFlow()
```

Comment:
What about channels? Are they good to use like this? As far as I understood they are better as a one time event than shared flows or other flows?